File: tests/testdata/codegen/using-object-as-input-for-any-pp/python/using-object-as-input-for-any.py
import pulumi
import pulumi_aws_native as aws_native

role = aws_native.iam.Role("role",
    role_name="ScriptIAMRole",
    assume_role_policy_document={
        "Version": "2012-10-17",
        "Statement": [{
            "Effect": "Allow",
            "Action": "sts:AssumeRole",
            "Principal": {
                "Service": [
                    "cloudformation.amazonaws.com",
                    "gamelift.amazonaws.com",
                ],
            },
        }],
    })

File: pkg/codegen/testing/utils/testdata/using-object-as-input-for-any-pp/dotnet/using-object-as-input-for-any.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using AwsNative = Pulumi.AwsNative;

return await Deployment.RunAsync(() => 
{
    var role = new AwsNative.Iam.Role("role", new()
    {
        RoleName = "ScriptIAMRole",
        AssumeRolePolicyDocument = new Dictionary<string, object?>
        {
            ["Version"] = "2012-10-17",
            ["Statement"] = new[]
            {
                new Dictionary<string, object?>
                {
                    ["Effect"] = "Allow",
                    ["Action"] = "sts:AssumeRole",
                    ["Principal"] = new Dictionary<string, object?>
                    {
                        ["Service"] = new[]
                        {
                            "cloudformation.amazonaws.com",
                            "gamelift.amazonaws.com",
                        },
                    },
                },
            },
        },
    });

});

